packages/code-utils/src/index.ts

```diff
@@ -1,53 +1,72 @@
 import parseOpcode from "./opcodes";
-import type { Instruction, OpcodeTable, opcodeObject, opcodes } from "./types";
-export type { Instruction, OpcodeTable, opcodeObject, opcodes };
+import type { Instruction, OpcodeTable, DisassemblyOptions } from "./types";
+export type { Instruction, OpcodeTable, DisassemblyOptions };
 import * as cbor from "cbor";
 
 /**
  * parseCode - return a list of instructions given a 0x-prefixed code string.
  *
- * If numInstructions is not passed in, we attempt to strip contract
- * metadata.  This won't work very well if the code is for a constructor or a
- * contract that can create other contracts, but it's better than nothing.
+ * The optional second options argument allows two options; both are ways of
+ * attempting to limit the disassembly to only the code section rather than the
+ * data section.  If maxInstructionCount is used, the disassembly will be limited
+ * to the specified number of instructions (one may pass in here the number of
+ * instructions in the corresponding source map).
  *
- * WARNING: Don't invoke the function that way if you're dealing with a
- * constructor with arguments attached!  Then you could get disaster!
+ * If attemptStripMetadata is used, we will attempt to strip the metadata at the
+ * end of the code.  This is not reliable, and should be avoided if better
+ * alternatives are available.  It may be particularly unreliable when dealing with
+ * constructors that have had arguments attached to the end!
  *
- * If you pass in numInstructions (hint: count the semicolons in the source
- * map, then add one) this is used to exclude metadata instead.
+ * These options can be combined, although I'm not sure why you'd want to.
  *
  * @param  {String} hexString Hex string representing the code
  * @return Array               Array of instructions
  */
 export function parseCode(
```

Comment:
How much do we care about cyclomatic complexity? This function had what I'd consider to be high cyclomatic complexity prior to this change, and then this change increases it substantially. Perhaps this should be broken out into a few child functions for easier readability?

Reply:
Perhaps we can save that for when this function will be need to be substantially rewritten come Shanghai? :)

Reply:
Same.